src/solve/fulfill.rs

```diff
@@ -312,24 +307,29 @@ impl<'s> Fulfill<'s> {
                             }
                         }
 
-                        solution.is_ambig()
+                        (solution.is_ambig(), solution.cannot_be_proven())
                     }
                     Obligation::Refute(ref goal) => {
-                        self.refute(goal)? == NegativeSolution::Ambiguous
+                        let answer = self.refute(goal)?;
+                        (answer == NegativeSolution::Ambiguous, answer == NegativeSolution::CannotProve)
                     }
                 };
 
                 if ambiguous {
                     debug!("ambiguous result: {:?}", obligation);
                     obligations.push(obligation);
                 }
+
+                // If one of the obligations cannot be proven, then the whole goal
+                // cannot be proven either.
```

Comment:
To make sure I understand: We don't stop immediately here because if we were to encounter a hard error somewhere else, that would "override" this cannot prove result, right?